examples/mysql_to_kafka.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+#
+# Output Kafka events to the console from MySQL replication stream
+#
+
+from pymysqlreplication import BinLogStreamReader
+from pymysqlreplication.row_event import (
+    DeleteRowsEvent,
+    UpdateRowsEvent,
+    WriteRowsEvent,
+)
+
+from kafka.admin import KafkaAdminClient, NewTopic
+from kafka import KafkaConsumer, KafkaProducer
+
+MYSQL_SETTINGS = {"host": "127.0.0.1", "port": 3306, "user": "root", "passwd": ""}
+
+
+def create_kafka_topic(topic_name, num_partitions=1, replication_factor=1):
+    admin_client = KafkaAdminClient(bootstrap_servers="127.0.0.1:9092")
+
+    topic = NewTopic(
+        name=topic_name,
+        num_partitions=num_partitions,
+        replication_factor=replication_factor,
+    )
+
+    admin_client.create_topics(new_topics=[topic])
+
+
+def main():
+    global message_body
+    producer = KafkaProducer(
+        bootstrap_servers="127.0.0.1:9092",
+        value_serializer=lambda v: str(v).encode("utf-8"),
+    )
+
+    stream = BinLogStreamReader(
+        connection_settings=MYSQL_SETTINGS,
+        server_id=3,
+        only_events=[DeleteRowsEvent, UpdateRowsEvent, WriteRowsEvent],
+    )
+
+    for binlogevent in stream:
+        for row in binlogevent.rows:
+            if isinstance(binlogevent, DeleteRowsEvent):
+                topic = "deleted"
+                message_body = row["values"].items()
+
+            elif isinstance(binlogevent, UpdateRowsEvent):
+                topic = "updated"
+                message_body = row["after_values"].items()
+
+            elif isinstance(binlogevent, WriteRowsEvent):
+                topic = "created"
+                message_body = row["values"].items()
+
+            producer.send(topic, key=None, value=dict(message_body))
+
+    consumer = KafkaConsumer(
+        "deleted",
+        "updated",
+        "created",
+        bootstrap_servers="127.0.0.1:9092",
+        value_deserializer=lambda x: x.decode("utf-8"),
+        auto_offset_reset="earliest",
+        group_id="1",
+    )
+
+    for message in consumer:
```

Comment:
Ok, I handle with KeyboardInterrupt.
Additional, I found one more error in process of closing like this.

<img width="1299" alt="스크린샷 2023-10-18 오후 1 47 28" src="https://github.com/julien-duponchelle/python-mysql-replication/assets/89684956/5f2e48cb-b378-4da0-9ead-93b3b559a3a3">

It is issue in Kafka-python https://github.com/dpkp/kafka-python/issues/1583
So I add time.sleep(1) under producer.close().
I think it's not the best way to handle this error. but I don't know how to handle this with other way.
Isn't this a problem?

Finally, I think we handle all the errors occur when we use this example.
Thx for your feedback and support.

Reply:
The purpose of example codes are to showcase the use case.
In this case, it shows how you can incorporate kafka-python.
Thanks to you, this is already fulfilled.
So any other details that are specific to the external dependency can be left out.
Going top deep with the exception may hinder newcomers' understanding of the examples.
Catching BaseException or KeyboardInterrupt IS NOT a perfect way to detecting end of the program
but it can hint the users that the code ends at that point and I think that's enough for the example code.
So great job!